code/modules/mob/living/carbon/human/species/station/station_special_abilities_vr.dm

```diff
@@ -80,55 +77,56 @@
-					brainloss += 10 //Reviving from dead means you take a lil' brainloss on top of whatever was healed in the revive.
-					return
-				else
-					return //Runtime prevention
-			else //don't have nutriment to hatch! Or you somehow died in between completing your revive and hitting hatch.
-				to_chat(src, "Your body was unable to regenerate, what few living cells remain require additional nutrients to complete the process.")
-				reviving = FALSE // so they can try again when they're given a kickstart
+	if(confirm == "Yes" && reviving)
+
+		//Dead when hatching
+		if(stat == DEAD)
+			//Check again for nutriment (necessary?)
+			if(hasnutriment())
+				chimera_hatch()
+				visible_message("<span class='danger'><p><font size=4>The lifeless husk of [src] bursts open, revealing a new, intact copy in the pool of viscera.</font></p></span>") //Bloody hell...
+				brainloss += 10 //Reviving from dead means you take a lil' brainloss on top of whatever was healed in the revive.
 				return
 
-		else if(stat != DEAD) //If they're alive at the time of regenerating.
+			//Don't have nutriment to hatch! Or you somehow died in between completing your revive and hitting hatch.
+			else
+				to_chat(src, "Your body was unable to regenerate, what few living cells remain require additional nutrients to complete the process.")
+				reviving = FALSE //So they can try again when they're given a kickstart
+
+		//Alive when hatching
+		else
 			chimera_hatch()
 			visible_message("<span class='danger'><p><font size=4>The dormant husk of [src] bursts open, revealing a new, intact copy in the pool of viscera.</font></p></span>") //Bloody hell...
-			return
-		else
-			return //Runtime prevention.
 
 /mob/living/carbon/human/proc/chimera_hatch()
-	nutrition -= nutrition/2 //Cut their nutrition in half.
-	var/old_nutrition = nutrition //Since the game is being annoying.
+	last_special = world.time + 1 HOUR
+	verbs -= /mob/living/carbon/human/proc/hatch
 	to_chat(src, "<span class='notice'>Your new body awakens, bursting free from your old skin.</span>")
-	var/T = get_turf(src)
-	new /obj/effect/gibspawner/human/scree(T)
-	var/braindamage = brainloss/2 //If you have 100 brainloss, it gives you 50.
-	does_not_breathe = 0 //start breathing again
-	revive() // I did have special snowflake code, but this is easier.
-	weakened = 2 //Not going to let you get up immediately. 2 ticks before you get up. Overrides the above 10000 weaken.
+	reviving = FALSE
+
+	//Modify and record values (half nutrition and braindamage)
+	var/old_nutrition = nutrition * 0.5
+	var/braindamage = brainloss * 0.5
+
+	//I did have special snowflake code, but this is easier.
+	revive()
 	mutations.Remove(HUSK)
 	nutrition = old_nutrition
-	brainloss = braindamage //Gives them half their prior brain damage.
-	update_canmove()
+	brainloss = braindamage
```

Comment:
Changelog shows brainloss was moved to the brain organ. Use setBrainLoss(braindamage).